modules/ai/module.php

```diff
@@ -316,6 +348,12 @@ public function ajax_ai_get_remote_config() {
 	}
 
 	private function verify_permissions( $editor_post_id ) {
+		$referer = wp_get_referer();
+
+		if ( str_contains( $referer, 'wp-admin/upload.php' ) && current_user_can( 'upload_files' ) ) {
+			return true;
+		}
+
```

Comment:
Please extract to `verify_upload_permissions` and use it in the relevant request

Reply:
it means adding it across 8 methods (all the image-related requests), isn't it better to keep it here instead of all the duplication? wdyt?

Reply:
It's better to be explicit.